core/src/main/java/com/sequenceiq/cloudbreak/conclusion/ConclusionCheckerService.java

```diff
@@ -0,0 +1,52 @@
+package com.sequenceiq.cloudbreak.conclusion;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.event.ResourceEvent;
+
+@Service
+public class ConclusionCheckerService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConclusionCheckerService.class);
+
+    @Inject
+    private ConclusionCheckerFactory conclusionCheckerFactory;
+
+    @Inject
+    private CloudbreakFlowMessageService flowMessageService;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    public void runConclusionChecker(Long resourceId, String eventType, ResourceEvent resourceEvent, String... eventMessageArgs) {
+        try {
+            ConclusionChecker conclusionChecker = conclusionCheckerFactory.getConclusionChecker(ConclusionCheckerType.DEFAULT);
+            ConclusionResult conclusionResult = conclusionChecker.doCheck(resourceId);
+            if (entitlementService.conclusionCheckerSendUserEventEnabled(ThreadBasedUserCrnProvider.getAccountId()) && conclusionResult.isFailureFound()) {
+                flowMessageService.fireEventAndLog(resourceId, eventType, resourceEvent, getFullEventMessageArgs(eventMessageArgs, conclusionResult));
+            }
```

Comment:
a bit out of context but do we really need to execute the check when for example the entitlement is not granted and we are not able to fire any event due to that?

Reply:
Yes, the entitlement is related only to the event sending, the checker will log the conclusion results anyway.